Add field aliasing

clinta opened the pull request:
<!--
Thanks for your contribution! Check out the
[contributing docs](https://github.com/Khan/genqlient/blob/main/docs/CONTRIBUTING.md)
for more on contributing to genqlient.
-->

  - Added @genqlient(alias) directive to customize field names without GraphQL aliases (fixes #367)

I have:
- [x] Written a clear PR title and description (above)
- [x] Signed the [Khan Academy CLA](https://www.khanacademy.org/r/cla)
- [x] Added tests covering my changes, if applicable
- [x] Included a link to the issue fixed, if applicable
- [x] Included documentation, for new features
- [x] Added an entry to the changelog

csilvers commented:
Hi @clinta , thank you for your contribution!  Can you separate the last item ("Fixed path handling in file globbing to properly normalize paths") into its own PR?  I suspect that will be easier/faster to get in than the others.

Ideally, you'd have three PRs, so we could consider the camel-case and the alias changes separately.  I know those might be more intertwined, but hopefully it's not that much work.

clinta commented:
That's my mistake on the changelog. I was reviewing and summarizing differences between main, but I was looking at my main which hadn't pulled recent changes, after I already rebased this branch. So when I was reviewing changes, that looked like a change that needed to be included in the changelog for this PR. I fixed that.

I should be able to split up the alias and snake_case option into separate PRs.

clinta commented:
This is now a focused PR just on adding the alias directive. I'll open another PR based on top of this with the snake_case conversion feature.

benjaminjkraft commented:
Thanks for splitting this out, it's very clean now! (I'm still sad we need it, but oh well, that's life being a client.)
